components/Activitylist.js

```diff
@@ -40,6 +49,33 @@ export default function Activitylist({
     }
   }
 
+  function handleAddFavoriteButton(id) {
+    const activitySuggestionCard = activityCards.find((card) => card.id === id);
+
+    if (activitySuggestionCard) {
+      setActivityCards(
+        activityCards.map((card) =>
+          card.id === id
+            ? {
+                ...card,
+                userData: card.userData.some((user) => user.userID === userID)
```

Comment:
userData does not imply any information how they are connected to the cards. I guess its the users that "like" an activity? 
Maybe call it `likedByUsers`?

Reply:
very good naming !